lib/fips_lookup.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require_relative "fips_lookup/version"
+require 'csv'
+require 'pathname'
+
+module FipsLookup
+  STATE_CODES = { "AL" => "01", "AK" => "02", "AZ" => "04", "AR" => "05", "CA" => "06", "CO" => "08",
+                  "CT" => "09", "DE" => "10", "DC" => "11", "FL" => "12", "GA" => "13", "HI" => "15",
+                  "ID" => "16", "IL" => "17", "IN" => "18", "IA" => "19", "KS" => "20", "KY" => "21",
+                  "LA" => "22", "ME" => "23", "MD" => "24", "MA" => "25", "MI" => "26", "MN" => "27",
+                  "MS" => "28", "MO" => "29", "MT" => "30", "NE" => "31", "NV" => "32", "NH" => "33",
+                  "NJ" => "34", "NM" => "35", "NY" => "36", "NC" => "37", "ND" => "38", "OH" => "39",
+                  "OK" => "40", "OR" => "41", "PA" => "42", "RI" => "44", "SC" => "45", "SD" => "46",
+                  "TN" => "47", "TX" => "48", "UT" => "49", "VT" => "50", "VA" => "51", "WA" => "53",
+                  "WV" => "54", "WI" => "55", "WY" => "56", "AS" => "60", "GU" => "66", "MP" => "69",
+                  "PR" => "72", "UM" => "74", "VI" => "78"
+                }.freeze
+
+  class << self
+    attr_accessor :county_fips
+
+    def county(state_param, county_name_param, return_nil = false)
+      lookup = [state_param.upcase, county_name_param.upcase]
```

Comment:
Ruby hash key can be an Array of strings, so using the method input as the strings within the array. so input like "michigan", "wayne county" would be stored in the memoization hash like:
`{["MICHIGAN", "WAYNE COUNTY"] => "26163"` 

Continue building the memoized hash `county_fips` with user input and it could end up storing the same value with a different lookup: 
`{["MI", "WAYNE COUNTY"] => "26163"}` 

should the state_param be found first and generalize the lookup to search for the state code first? (instead of below in line 29)

Reflecting while typing this out (^) I might be leaning towards finding the state code first within the `.county` method and using that throughout instead of passing along the state param. Then the `county_lookup` method that does the work can be made private and accessed only if the state code can be found and is searchable.

Reply:
Changing with next PR (named_args branch) https://github.com/3barroso/fips_lookup/pull/3 to have `.county` method lookup the state code before any other work happens. using found state_code in `county_fips[lookup]`